### DISCUSSION TITLE
How to get the probability of each model?

### DISCUSSION THREAD

**Discussion by plusante** [mboquien/cigale#102] [Q&A]
Hi! I am new to Cigale and am currently trying to perform SED fitting to determine the stellar mass of some galaxies. I have input some parameters and obtained around 4000 models. I want to know the probability of each model so that I can adjust my parameters for the next run to focus on the most probable settings. I tried to read the *chi2*.npy file by np.memmap(...), but it only gives a list of chi-square values, and I don't know which models they correspond to. Does anyone have any idea? Thanks in advance.

**Comment by Guang91** [comment]
We have a python code (associated with the manual) that may do the job you want:
https://gitlab.lam.fr/cigale/manual/-/blob/master/code/read_chi.py?ref_type=heads

**Reply by plusante** [reply]
I will check it out. Thank you!

**Reply by mboquien** [reply]
Note that in the newest versions the npy are just regular numpy files. Irrespective of this, you can just load the npy file for each physical parameter. Then you just need to retrieve the specific combination of parameters.
